### PR TITLE
Move 'o nas' static page to the new Angular frontend #635

### PR DESCRIPTION
**Pull request by TrebuhD**
Move 'o nas' static page to the new Angular frontend #635 #642

__Story / Bug id:__ https://github.com/CodeForPoznan/volontulo/issues/635

__Reference person:__ @TrebuhD

__Description:__

Moved the 'o-nas' static page to the new Angular frontend.

__New imports / dependencies:__

* none

### CODE REVIEW

File: frontend/src/app/app.module.ts [hunk 1]
```diff
@@ -11,13 +11,18 @@ import { WindowService, WindowFactory } from './window.service';
...
 import { HeaderComponent } from './header/header.component';
+import { AboutUsComponent } from './static/about-us.component';
 
 const appRoutes: Routes = [
   // {
   //   path: '',
   //   component: HomeComponent
   // },
   {
+    path: 'about-us',
```
Comment (magul) [inline frontend/src/app/app.module.ts]: This page on old Django was available under `o-nas` url. Please keep it that way.

File: frontend/src/app/footer/footer.component.html [hunk 2]
```diff
@@ -9,7 +9,7 @@ <h3 class="nav-title">Informacje</h3>
             <a class="nav-link" href="#">Strona główna</a>
             <a class="nav-link" href="#">Załóż konto</a>
             <a class="nav-link" href="#">Kontakt z administratorem</a>
-            <a class="nav-link" href="#">O nas</a>
+            <a class="nav-link" href="about-us">O nas</a>
```
Comment (magul) [inline frontend/src/app/footer/footer.component.html]: `o-nas` intead of `about-us`

Comment (magul) [reply frontend/src/app/footer/footer.component.html]: + You need to change routing on old Django-based footer (see for setting `ANGULAR_ROOT`)

File: frontend/src/app/app.component.html [hunk 3]
```diff
@@ -1 +1,7 @@
-<router-outlet></router-outlet>
+<volontulo-header></volontulo-header>
```
Comment (magul) [inline frontend/src/app/app.component.html]: I know, that we were waiting for `ng test` fix (and it landed in `master`, so You're welcome to check it) but if You're moving `<volontulo-header>` and `<volontulo-footer>` here, the test in `app.component.spec.ts` will fail with:

```
	'volontulo-header' is not a known element:
	1. If 'volontulo-header' is an Angular component, then verify that it is part of this module.
	2. If 'volontulo-header' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message. ("[ERROR ->]<volontulo-header></volontulo-header>
	<div class="container">
	  <div class="main-content mt-3">
	"): ng:///DynamicTestModule/AppComponent.html@0:0
	'volontulo-footer' is not a known element:
	1. If 'volontulo-footer' is an Angular component, then verify that it is part of this module.
	2. If 'volontulo-footer' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message. ("
	  </div>
	</div>
	[ERROR ->]<volontulo-footer></volontulo-footer>
	"): ng:///DynamicTestModule/AppComponent.html@6:0
	Error: Template parse errors:
	    at syntaxError (webpack:///~/@angular/compiler/@angular/compiler.es5.js:1694:21 <- src/test.ts:28285:34)
	    at TemplateParser.Array.concat.TemplateParser.parse (webpack:///~/@angular/compiler/@angular/compiler.es5.js:12932:0 <- src/test.ts:39523:19)
	    at JitCompiler.Array.concat.JitCompiler._compileTemplate (webpack:///~/@angular/compiler/@angular/compiler.es5.js:27126:0 <- src/test.ts:53717:39)
	    at webpack:///~/@angular/compiler/@angular/compiler.es5.js:27045:47 <- src/test.ts:53636:62
	    at Set.forEach (<anonymous>)
	    at JitCompiler.Array.concat.JitCompiler._compileComponents (webpack:///~/@angular/compiler/@angular/compiler.es5.js:27045:0 <- src/test.ts:53636:19)
	    at webpack:///~/@angular/compiler/@angular/compiler.es5.js:26946:0 <- src/test.ts:53537:19
	    at Object.then (webpack:///~/@angular/compiler/@angular/compiler.es5.js:1683:32 <- src/test.ts:28274:148)
	    at JitCompiler.Array.concat.JitCompiler._compileModuleAndAllComponents (webpack:///~/@angular/compiler/@angular/compiler.es5.js:26944:0 <- src/test.ts:53535:26)
	    at JitCompiler.Array.concat.JitCompiler.compileModuleAndAllComponentsAsync (webpack:///~/@angular/compiler/@angular/compiler.es5.js:26876:0 <- src/test.ts:53467:37)
	'volontulo-header' is not a known element:
	1. If 'volontulo-header' is an Angular component, then verify that it is part of this module.
	2. If 'volontulo-header' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message. ("[ERROR ->]<volontulo-header></volontulo-header>
	<div class="container">
	  <div class="main-content mt-3">
	"): ng:///DynamicTestModule/AppComponent.html@0:0
	'volontulo-footer' is not a known element:
	1. If 'volontulo-footer' is an Angular component, then verify that it is part of this module.
	2. If 'volontulo-footer' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message. ("
	  </div>
	</div>
	[ERROR ->]<volontulo-footer></volontulo-footer>
	"): ng:///DynamicTestModule/AppComponent.html@6:0
	Error: Template parse errors:
	    at syntaxError (webpack:///~/@angular/compiler/@angular/compiler.es5.js:1694:21 <- src/test.ts:28285:34)
	    at TemplateParser.Array.concat.TemplateParser.parse (webpack:///~/@angular/compiler/@angular/compiler.es5.js:12932:0 <- src/test.ts:39523:19)
	    at JitCompiler.Array.concat.JitCompiler._compileTemplate (webpack:///~/@angular/compiler/@angular/compiler.es5.js:27126:0 <- src/test.ts:53717:39)
	    at webpack:///~/@angular/compiler/@angular/compiler.es5.js:27045:47 <- src/test.ts:53636:62
	    at Set.forEach (<anonymous>)
	    at JitCompiler.Array.concat.JitCompiler._compileComponents (webpack:///~/@angular/compiler/@angular/compiler.es5.js:27045:0 <- src/test.ts:53636:19)
	    at webpack:///~/@angular/compiler/@angular/compiler.es5.js:26946:0 <- src/test.ts:53537:19
	    at Object.then (webpack:///~/@angular/compiler/@angular/compiler.es5.js:1683:32 <- src/test.ts:28274:148)
	    at JitCompiler.Array.concat.JitCompiler._compileModuleAndAllComponents (webpack:///~/@angular/compiler/@angular/compiler.es5.js:26944:0 <- src/test.ts:53535:26)
	    at JitCompiler.Array.concat.JitCompiler.compileModuleAndAllComponentsSync (webpack:///~/@angular/compiler/@angular/compiler.es5.js:26868:0 <- src/test.ts:53459:42)
```

Comment (magul) [reply frontend/src/app/app.component.html]: It's a thing that @SabinaCzopik should resolve in her pull request, but if She will procrastinate, You're branch will land in `master` and automatically resolve her issue.

File: frontend/src/app/app.component.html [hunk 4]
```diff
@@ -1 +1,7 @@
...
+<div class="container">
```
Comment (magul) [inline frontend/src/app/app.component.html]: I don't know if it should be here, as on homepage in this part of page is a baner, that spread on the whole width.

File: frontend/src/app/static/about-us.component.html [hunk 5]
```diff
@@ -0,0 +1,57 @@
+<div class="container my-5">
+  <h1 class="my-3">O nas</h1>
+  <div>
+    <div class="paragraph">Lubisz nowe wyzwania? Chcesz pomagać innym? Zależy Ci na rozwoju? Szukasz wolontariuszy?
+      Jeśli
```
Comment (magul) [inline frontend/src/app/static/about-us.component.html]: What's a reason of this wired formatting? Can You make it more like similar line length paragraphs?

File: frontend/src/app/app.module.ts (hunk 1 above)
Comment (TrebuhD) [reply frontend/src/app/app.module.ts]: It was under `about-us`.

File: frontend/src/app/app.component.html (hunk 3 above)
Comment (magul) [reply frontend/src/app/app.component.html]: I see, that error changed to
```
	Failed: Response with status: 0  for URL: null
	    at webpack:///~/@angular/core/@angular/core/testing.es5.js:50:0 <- src/test.ts:26775:26
	    at webpack:///~/@angular/core/@angular/core/testing.es5.js:101:0 <- src/test.ts:26826:17
	    at ZoneDelegate.invoke (webpack:///~/zone.js/dist/zone.js:392:0 <- src/polyfills.ts:1610:26)
	    at AsyncTestZoneSpec.Array.concat.AsyncTestZoneSpec.onInvoke (webpack:///~/zone.js/dist/async-test.js:49:0 <- src/test.ts:69542:39)
```
if means basically, that there is a http request, that's not successfully resolved - the only http request right now, that is generated by our Angular is request for current user made by `AuthService`.

To resolve this issue, basically You need to mock `AuthService` - how to do this? see: https://blog.thoughtram.io/angular/2016/11/28/testing-services-with-http-in-angular-2.html

File: backend/apps/volontulo/context_processors.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+
+"""
+.. module:: authentication
+"""
+
+from django.conf import settings
+
+
+def angular_root(request):
```
[the remark this answers is not in the document]
Comment (TrebuhD) [reply backend/apps/volontulo/context_processors.py]: It fails without this argument.

**Pull request merged**